posthog/api/plugin.py

```diff
@@ -0,0 +1,132 @@
+import json
+from typing import Any, Dict
+
+import requests
+from django.conf import settings
+from rest_framework import request, serializers, viewsets
+from rest_framework.decorators import action
+from rest_framework.exceptions import APIException
+from rest_framework.response import Response
+
+from posthog.models import Plugin, PluginConfig
+from posthog.plugins import download_plugin_github_zip, reload_plugins_on_workers
+
+
+class PluginSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Plugin
+        fields = ["id", "name", "description", "url", "config_schema", "tag", "from_json"]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            raise APIException("Plugin installation via the web is disabled!")
```

Comment:
I would change this to a `BadRequest`, this will return a 500 error code when it's not really the case.

Reply:
I converted all `APIException`s to `ValidationError`s in this file. Now they give 400 codes:
![image](https://user-images.githubusercontent.com/53387/97705921-2493c980-1ab5-11eb-9a4c-59344400c901.png)